shell/utils/fleet.ts

```diff
@@ -0,0 +1,106 @@
+import { BundleDeploymentResource, ModifiedStatus, NonReadyBundle } from '@shell/types/resources/fleet';
+import { STATES_ENUM } from '@shell/plugins/dashboard-store/resource-class';
+import { FLEET as FLEET_ANNOTATIONS } from '@shell/config/labels-annotations';
+
+type state = STATES_ENUM.READY | STATES_ENUM.MODIFIED | STATES_ENUM.MISSING | STATES_ENUM.ORPHANED;
+
+interface Resource extends BundleDeploymentResource {
+  state: state,
+}
+
+interface BundleDeploymentStatus {
+  resources?: BundleDeploymentResource[],
+  modifiedStatus?: ModifiedStatus[],
+}
+
+interface BundleStatusSummary {
+  nonReadyResources?: NonReadyBundle[],
+}
+
+interface BundleStatus {
+  resourceKey?: BundleDeploymentResource[],
+  summary?: BundleStatusSummary,
+}
+
+// bundleDeploymentResources extracts the list of resources deployed by a BundleDeployment
+export function bundleDeploymentResources(status: BundleDeploymentStatus): Resource[] {
```

Comment:
: Resource[]  isn't right, it's more like `(Resource | ModifiedStatus) []`, however that makes usages of this a bit odd. it can always be a new type.

Generally though there's lots of TS errors that need to be resolved further down. happy to work together to resolve them?

Reply:
I think I've sorted this out, my intention is that `Resource` is some sort of new type according to what the caller needs, not tied to the Fleet description.